src/backend/mock.ts

```diff
@@ -16,16 +16,13 @@ import Logging from '@/utils/logging';
 
 const console = Logging.mock;
 
-export default class MockBackend extends events.EventEmitter implements KubernetesBackend, VMExecutor {
-  readonly kubeBackend = this;
+export default class MockBackend extends events.EventEmitter implements VMExecutor {
+  readonly kubeBackend = new MockKubernetesBackend();
```

Comment:
Might be better to inject this as a dependency rather than constructing it in in the class? I guess this is dependent on what kind of testing we are doing. I'm sure it's fine for now.

Reply:
Testing in this area is currently… anemic? It's mostly covered by end-to-end tests.

After the first round of refactoring is done, we should be able to split the kubernetes backends out to separate files, at which point we can revisit and pass this in to get better testing.